### DISCUSSION TITLE
Periodic and reflected index wrapping

### DISCUSSION THREAD

**Discussion by jgostick** [Quansight-Labs/ndindex#179] [Ideas]
I guess this is more of a feature request than an idea, but I have always wished I could tell numpy to wrap or reflect indices that go out of range. My particular use case is performing random walk simulations on 3D domains FWIW. I feel like your approach here could enable this trick pretty easily, so thought I'd put it on your radar if it's not already.  And while I'm at it, might as well add a truncate mode as well, so if I ask for a sub-block of a large array it will return a truncated block if my requested block was beyond the bounds.

**Comment by asmeurer** [comment]
Wrapping and truncation (I would call it clipping, since it's like the NumPy `clip` function) would be useful. Reflection is interesting, but I'm not sure if it's as common.

One problem is that these operations wouldn't preserve slices in general, meaning you might have to convert a slice into an integer index.
